lib/src/main/java/com/cloudinary/android/preprocess/Transcode.java

```diff
@@ -0,0 +1,144 @@
+package com.cloudinary.android.preprocess;
+
+import android.content.Context;
+import android.media.MediaCodecInfo;
+import android.media.MediaExtractor;
+import android.media.MediaFormat;
+import android.net.Uri;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.linkedin.android.litr.MediaTransformer;
+import com.linkedin.android.litr.TransformationListener;
+import com.linkedin.android.litr.analytics.TrackTransformationInfo;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Preprocess implementation for video transcoding.
+ */
+public class Transcode implements Preprocess<Uri>, TransformationListener {
```

Comment:
This class should not implement `TransformationListener` itself since it exposes the LiTr classes directly, and will break API if we switch to a different library or implement our own solution in the future.
A private variable (perhaps using an  inner class) can be the listener and change the state accordingly.